### ISSUE TITLE
Update VMMC vs HIV testing code ordering

### ISSUE THREAD

**Issue by pineapple-cat** [UCL/hivpy#185]
It would be more logical, with respect to the timeline of events in reality, to update VMMC after HIV testing so that people about to undergo VMMC are tested first. This way we can rely on testing and diagnosis for the current time step when selecting the VMMC sub-population. This would also mean we need to somehow mark people for VMMC before they go through the procedure and that VMMC does not affect sexual behaviour until the next time step.

**Comment by andrew-phillips-1** [comment]
Yes I agree, although I don't think VMMC affects sexual behaviour, but if course it HIV acquisition risk.
